dsf/org.eclipse.cdt.dsf.ui/src/org/eclipse/cdt/dsf/debug/internal/ui/disassembly/DisassemblyPart.java

```diff
@@ -2126,8 +2126,10 @@ private void resetViewer() {
 		fPCHistory.clear();
 		fPendingPCUpdates.clear();
 		fFile2Storage.clear();
-		fDocument.clear();
+		var oldDocument = fDocument;
```

Comment:
I don't know if it is necessary to delay the dispose call until after the setDocument. On simple tests it works either way, but because the old document is passed to listeners in `fViewer.setDocument` call, it seems reasonable to leave it not disposed until the setDocument returns.

Reply:
Thanks @jonahgraham ,
I've ported your fix to my fork and it appears to work well.

I am also able to make the vanilla CDT Disassembly view fault using the approaches you describe.